classes/core/Controllers/Middlewares/DecodingApiToken.php

```diff
@@ -0,0 +1,101 @@
+<?php
+
+declare(strict_types=1);
+
+namespace PKP\core\Controllers\Middlewares;
+
+use APP\facades\Repo;
+use Closure;
+use DomainException;
+use Exception;
+use Firebase\JWT\JWT;
+use Firebase\JWT\SignatureInvalidException;
+use Illuminate\Auth\Access\AuthorizationException;
+use Illuminate\Http\JsonResponse;
+use Illuminate\Http\Response;
+
+use PKP\config\Config;
+use UnexpectedValueException;
+
+class DecodingApiToken
+{
+    /**
+     * Handle an incoming request.
+     *
+     * @param \Illuminate\Http\Request $request
+     *
+     */
+    public function handle($request, Closure $next)
+    {
+        $jwt = $request->query('apiToken');
+
+        if (!$jwt) {
+            /**
+             * If we don't have a token, it's for the authentication logic to handle if it's a problem.
+             */
+
+            return $next($request);
+        }
+
+        $secret = Config::getVar('security', 'api_key_secret', null);
+        if (!$secret) {
+            throw new AuthorizationException('api.500.apiSecretKeyMissing');
+        }
+
+        $apiUser = null;
+
+        try {
+            $apiToken = (string) JWT::decode($jwt, $secret, ['HS256']);
+
+            /**
+             * Compatibility with old API keys
+             *
+             * @link https://github.com/pkp/pkp-lib/issues/6462
+             */
+            if (substr($apiToken, 0, 2) === '""') {
+                $apiToken = json_decode($apiToken);
+            }
+
+            $apiUser = Repo::user()->getByApiKey($apiToken);
+
+            if ($apiUser === null || !$apiUser->getData('apiKeyEnabled')) {
+                return new JsonResponse(
+                    ['error' => 'api.403.unauthorized'],
+                    Response::HTTP_UNAUTHORIZED
+                );
+            }
+
+            $contextId = $request->attributes->get('pkpContext')->getData('id');
```

Comment:
These next two lines are dead code, I think?

Reply:
No, the `$apiUser->getRoles($contextId, true);` is used here to setting the user's roles, based on the given context.

See https://github.com/pkp/pkp-lib/blob/main/classes/user/User.inc.php#L471